core/ipc/sink/aws-sqs-impl/src/main/java/org/opennms/core/ipc/sink/aws/sqs/AmazonSQSRemoteMessageDispatcherFactory.java

```diff
@@ -0,0 +1,183 @@
+ */
+public class AmazonSQSRemoteMessageDispatcherFactory extends AbstractMessageDispatcherFactory<String> {
+
+    /** The Constant LOG. */
+    private static final Logger LOG = LoggerFactory.getLogger(AmazonSQSRemoteMessageDispatcherFactory.class);
+
+    /** The AWS configuration. */
+    private final Properties awsConfig = new Properties();
+
+    /** The configuration administration object. */
+    private ConfigurationAdmin configAdmin;
+
+    /** The reporter. */
+    private JmxReporter reporter;
+
+    /** The SQS Object. */
+    private AmazonSQS sqs;
+
+    /* (non-Javadoc)
+     * @see org.opennms.core.ipc.sink.common.AbstractMessageDispatcherFactory#getModuleMetadata(org.opennms.core.ipc.sink.api.SinkModule)
+     */
+    @Override
+    public <S extends Message, T extends Message> String getModuleMetadata(final SinkModule<S, T> module) {
+        return AmazonSQSUtils.getQueueName(module);
+    }
+
+    /* (non-Javadoc)
+     * @see org.opennms.core.ipc.sink.common.AbstractMessageDispatcherFactory#dispatch(org.opennms.core.ipc.sink.api.SinkModule, java.lang.Object, org.opennms.core.ipc.sink.api.Message)
+     */
+    @Override
+    public <S extends Message, T extends Message> void dispatch(SinkModule<S, T> module, String topic, T message) {
+        try (MDCCloseable mdc = Logging.withPrefixCloseable(MessageConsumerManager.LOG_PREFIX)) {
+            LOG.trace("dispatch({}): sending message {}", topic, message);
+            final String queueName = getModuleMetadata(module);
+            final String queueUrl = getQueueUrl(queueName);
+            if (queueUrl == null) {
+                LOG.warn("Cannot obtain URL for queue {}. The message cannot be sent.", queueName);
+            } else {
+                sqs.sendMessage(queueUrl, module.marshal((T)message));
+            }
+        }
+    }
+
+    /**
+     * Initializes the producers.
+     *
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    public void init() throws IOException {
+        try (MDCCloseable mdc = Logging.withPrefixCloseable(MessageConsumerManager.LOG_PREFIX)) {
+            registerJmxReporter();
+
+            // Defaults
+            awsConfig.clear();
+
+            // Retrieve all of the properties from org.opennms.core.ipc.sink.aws.cfg
+            final Dictionary<String, Object> properties = configAdmin.getConfiguration(AmazonSQSSinkConstants.AWS_CONFIG_PID).getProperties();
+            if (properties != null) {
+                final Enumeration<String> keys = properties.keys();
+                while (keys.hasMoreElements()) {
+                    final String key = keys.nextElement();
+                    awsConfig.put(key, properties.get(key));
+                }
+            }
+
+            LOG.info("AwsRemoteMessageDispatcherFactory: initializing the AWS SQS producer with: {}", awsConfig);
+            final ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
I believe this was a class-loader hack that we needed for Kafka, it may not be necessary here.

Reply:
ok